include/dlaf/matrix.tpp

```diff
@@ -47,29 +50,17 @@ Matrix<T, device>::Matrix(matrix::Distribution&& distribution, const matrix::Lay
 }
 
 template <class T, Device device>
-Matrix<T, device>::Matrix(matrix::Distribution&& distribution, const matrix::LayoutInfo& layout,
-                          ElementType* ptr)
+Matrix<T, device>::Matrix(Distribution&& distribution, const LayoutInfo& layout, ElementType* ptr)
     : Matrix<const T, device>(std::move(distribution), layout, ptr) {}
 
 template <class T, Device device>
-Matrix<T, device>::Matrix(const matrix::LayoutInfo& layout, ElementType* ptr)
+Matrix<T, device>::Matrix(const LayoutInfo& layout, ElementType* ptr)
     : Matrix<const T, device>(layout, ptr) {}
 
 template <class T, Device device>
 hpx::future<Tile<T, device>> Matrix<T, device>::operator()(const LocalTileIndex& index) noexcept {
-  std::size_t i = tileLinearIndex(index);
-  hpx::future<TileType> old_future = std::move(tile_futures_[i]);
-  hpx::promise<TileType> p;
-  tile_futures_[i] = p.get_future();
-  tile_shared_futures_[i] = {};
-  return old_future.then(hpx::launch::sync, [p = std::move(p)](hpx::future<TileType>&& fut) mutable {
-    try {
-      return std::move(fut.get().setPromise(std::move(p)));
```

Comment:
Instead of `std::move(fut.get().setPromise(std::move(p)));` why not `fut.get().setPromise(std::move(p));` the latter is already a rvalue.